src/proto/streams/streams.rs

```diff
@@ -1018,6 +1022,13 @@ impl OpaqueStreamRef {
             .recv
             .release_capacity(capacity, &mut stream, &mut me.actions.task)
     }
+
+    pub fn stream_id(&self) -> StreamId {
+        self.inner.lock()
+            .unwrap()
```

Comment:
I thought `unwrap`ping this was probably reasonable, rather than returning a `Result`, since if the mutex on the stream store is poisoned, we're in a pretty bad state (and the other functions on `StreamRef` also `unwrap`).